### DISCUSSION TITLE
Pivot + Shooter subsystem(s)

### DISCUSSION THREAD

**Discussion by GalexY727** [Patribots4738/Crescendo2024#22] [Ideas]
Should the pivot on the shooter and the actual shooter be the same subsystem? I feel that they would go smoothly together, and if we use SpeedAnglePairs it feels like it would be easy to integrate into one subsystem. Are there downsides to using only one subsystem rather than two, and should we look into splitting the two up? 

Also, should the rollers on the trap mechanism also be separate, or is it different than the pivot and shooter?

**Comment by jacueblol** [comment]
In my opinion there is a difference in between the two shooters, and I do think there should be some separation in code. However I do think they could work together very nice. I think a general class for the entire handoff mechanism, which includes the shooter for the pivot, the pivot and the shooter itself should be contained within one file, however shouldnt be necessarily be in one class, but instead 3 classes that all have access to any variables being used within the file. Let me know what you think @GalexY727

**Reply by jacueblol** [reply]
If possible I'd like to get a team on this today (along with one of us to overview it), just to see what it would look like.

**Reply by GalexY727** [reply]
ooh good thoughts, nested classes isn't an awful idea. We will see today what works best though.

**Comment by PatribotsProgramming** [comment]
Let's make this real, it's time to make the [drawing board](https://www.tldraw.com/r/Zm7S1et7eVf8pw5SzmG1k?viewport=1063,-126,556,266&page=page:page)

**Comment by jacueblol** [comment]
This was resolved as well as shown in the implementation of:
Shooting Calc Comand #33
Shooter Subystem #31
Trigger Subsystem #30
Claw Subsystem #29
Elevator Command #28
Elevator Subsystem #26
Pivot Subsystem #25

**Comment by jacueblol** [comment]
We still need to determine the best way to handle the handoff #27 so this will be reopened for that discussion

**Comment by GalexY727** [comment]
@Jacob1010-h can you make an announcement with all of the drawing board links we have so far and what they represent? Like a quick access kind of thing.

**Reply by jacueblol** [reply]
of course, ill add that to my todo, and if I don't do it during this period, ill do it tonight

**Comment by PatribotsProgramming** [comment]
There has been an update to the drawing board, as the motors in the robot have "shifted" and are now in a more finalized variant. [Check it out](https://www.tldraw.com/r/Zm7S1et7eVf8pw5SzmG1k?viewport=-314,-139,2010,1035&page=page:page)
